custom_components/tahoma/__init__.py

```diff
@@ -222,11 +227,11 @@ def log_device(message: str, device: Device) -> None:
 async def _block_if_core_is_configured(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     overkiz_config_entries = hass.config_entries.async_entries("overkiz")
 
-    for overkiz_entry in overkiz_config_entries:
-        if (
-            entry.data[CONF_USERNAME] == overkiz_entry.data[CONF_USERNAME]
+    return any(
+        (
+            overkiz_entry.source in [SOURCE_USER, SOURCE_ZEROCONF, SOURCE_DHCP]
```

Comment:
```suggestion
            overkiz_entry.source in [SOURCE_USER, SOURCE_ZEROCONF, SOURCE_DHCP]
```

When you set up an entry (even when clicked via zeroconf/dhcp), doesn't it always use source user? Wouldn't this current change detect if a hub has been discovered but not set up as well?

Reply:
From my testing, hubs that haven't been set up ar not in `hass.config_entries.async_entries("overkiz")`. And the source is kept as `zeroconf` (or `dhcp` I guess but untested) if you set up a hub that was auto discovered. I think to get `user`, you need to actively select "Add Integration" then "No, set up another instance of Overkiz (by Somfy)".

Reply:
![image](https://user-images.githubusercontent.com/5701372/152505333-3a1e0505-5dc2-4440-aa4e-0037a00d709c.png)

Reply:
Thanks! It is always a bit vague how that behaves, thanks for double checking.